data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,61 +71,69 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
   }
 
   @Override
   public void processUpgrade(String oldVersion, String newVersion) {
     long startupTime = System.currentTimeMillis();
     LOG.info("Start upgrade of published news displayed property");
-
-    boolean transactionStarted = false;
-    RequestLifeCycle.begin(container);
+    ExoContainerContext.setCurrentContainer(container);
+    List<String> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0).stream()
+        .map(newsTargetMetadata -> String.valueOf(newsTargetMetadata.getId()))
+        .collect(Collectors.toList());
+    List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
+    RequestLifeCycle.begin(entityManagerService);
     EntityManager entityManager = entityManagerService.getEntityManager();
     try {
-      if (!entityManager.getTransaction().isActive()) {
-        entityManager.getTransaction().begin();
-        transactionStarted = true;
-      }
-      List<Metadata> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0);
-
-      List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
-      for (Metadata newsTargetMetadata : newsTargetMetadatas) {
-        StringBuilder newsTargetsMetadataItemsQueryStringBuilder = new StringBuilder("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID = '");
-        newsTargetsMetadataItemsQueryStringBuilder.append(newsTargetMetadata.getId() + "'");
-        Query newsTargetsMetadataItemsQuery = entityManager.createNativeQuery(newsTargetsMetadataItemsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        newsTargetsMetadataItems.addAll(newsTargetsMetadataItemsQuery.getResultList());
+      Query getNewsTargetMetadataItemsQuery = entityManager.createNativeQuery("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID IN :newsTargetMetadatas", MetadataItemEntity.class);
+      getNewsTargetMetadataItemsQuery.setParameter("newsTargetMetadatas", newsTargetMetadatas);
+      newsTargetsMetadataItems = getNewsTargetMetadataItemsQuery.getResultList();
+    } catch (Exception e) {
+      throw new PersistenceException("Error when getting news target metadata items", e);
+    } finally {
+      RequestLifeCycle.end();
+    }
+    LOG.info("Total number of published news to be migrated: {}", newsTargetsMetadataItems.size());
+    int notMigratedPublishedNewsCount = 0;
+    for (List<MetadataItemEntity> newsTargetsMetadataItemsChunk : ListUtils.partition(newsTargetsMetadataItems, 10)) {
+      try {
+        int notMigratedPublishedNewsCountByTransaction = manageNewsTargetsMetadataItemsProps(newsTargetsMetadataItemsChunk);
+        LOG.info("{}/{} published news migrated", migratedPublishedNewsCount, newsTargetsMetadataItems.size());
+        notMigratedPublishedNewsCount += notMigratedPublishedNewsCountByTransaction;
+      } catch (Exception e) {
+        throw new PersistenceException("Error when managing news target metadata items props", e);
```

Comment:
It would be better to avoid using this specific exception used by JPA API and use another generic unchecked exception (if you don't want to define a new special Exception Class) like `IllegalStateException`